app/services/hyrax/collections/nested_collection_query_service.rb

```diff
@@ -0,0 +1,66 @@
+module Hyrax
+  module Collections
+    module NestedCollectionQueryService
+      # @api public
+      #
+      # What possible collections can nested within the given parent collection?
+      #
+      # @param parent [Collection]
+      # @param scope [Object] Typically a controller object that responds to `repository`, `can?`, `blacklight_config`, `current_ability`
+      # @param limit_to_id [nil, String] Limit the query to just check if the given id is in the response. Useful for validation.
+      # @return [Array<SolrDocument>]
+      def self.available_child_collections(parent:, scope:, limit_to_id: nil)
+        return [] unless parent.try(:nestable?)
+        return [] unless scope.can?(:edit, parent)
+        query_solr(collection: parent, access: :read, scope: scope, limit_to_id: limit_to_id)
+      end
+
+      # @api public
+      #
+      # What possible collections can the given child be nested within?
+      #
+      # @param child [Collection]
+      # @param scope [Object] Typically a controller object that responds to `repository`, `can?`, `blacklight_config`, `current_ability`
+      # @param limit_to_id [nil, String] Limit the query to just check if the given id is in the response. Useful for validation.
+      # @return [Array<SolrDocument>]
+      def self.available_parent_collections(child:, scope:, limit_to_id: nil)
+        return [] unless child.try(:nestable?)
+        return [] unless scope.can?(:read, child)
+        query_solr(collection: child, access: :edit, scope: scope, limit_to_id: limit_to_id)
+      end
```

Comment:
👍

Reply:
Wondering if there could be a filter to remove any collections from the potential list of parents and children that is already an ancestor or descendant of the parent/child?